src/Faker/Provider/Internet.php

```diff
@@ -331,4 +331,128 @@ public static function macAddress()
 
         return $mac;
     }
+
+    /**
+     *
+     */
+    const SMALLEST_HTML_LENGTH = 4; // <br>
+
+    /**
+     * @param string $tag
+     * @param string $content
+     * @param string $attributes as a string
+     *
+     * @return string
+     */
+    protected static function buildHtmlTag($tag, $content = '')
+    {
+        return sprintf('<%s>%s</%s>', $tag, $content, $tag);
+    }
+
+    /**
+     * @return array of distinct block-level and inline-level elements names
+     */
+    protected static function getHtmlTags()
+    {
+        return array(
+            "block" =>  array("h1", "h2", "h3", "p"),
+            "inline" => array("i", "b", "span")
+        );
+    }
+
+    /**
+     * @link http://w3c.github.io/html-reference/global-attributes.html
+     * pro-memoria for now
+     *
+     * @return array
+     */
+    protected static function getRandomHtmlTagsGlobalAttribute()
+    {
+        return self::randomElement(array(
```

Comment:
The right way to do that in Faker is to use a list of possible formats, pick one random format, then call `$generator->parse($format)`. Something like:

```php
protected static $htmlAttributeFormats = array(
  'accesskey="{{randomletter}}"',
  'class="{{word}}"',
  'contenteditable="true"',
  'contenteditable="false"',
  // etc
);
```

And if the `parse()` method seems too limited because it doesn't allow formatters with arguments... improve it!